slack_sdk/web/client.py

```diff
@@ -3143,9 +3143,10 @@ def files_upload_v2(
                 channel_to_share = channels[0]
         completion = self.files_completeUploadExternal(
             files=[{"id": f["file_id"], "title": f["title"]} for f in files],
-            channel=channel_to_share,
+            channel_id=channel_to_share,
             initial_comment=initial_comment,
             thread_ts=thread_ts,
+            token=kwargs.get("token"),
```

Comment:
While fixing the issue, I've noticed that this one is missing here. With this, developers can pass "token" to v2 method to overwrite the token given when initializing `WebClient`. Current code does not response the parameter for this specific API call. That was a different bug.